### PR TITLE
fixed response if user isn't found

### PR DESCRIPTION
**Pull request by artem-zhukov**
connect #40

### CODE REVIEW

File: src/main/java/juja/microservices/users/service/UserService.java [hunk 1]
```diff
@@ -59,9 +59,15 @@ private UserDTO convertGetAllUsersDto(User user) {
...
-                .map(this::convertGetUuidBySlackDto)
-                .collect(Collectors.toList());
+        List<UserDTO> result;
+        try {
+            result = users.stream()
+                    .map(this::convertGetUuidBySlackDto)
+                    .collect(Collectors.toList());
+        } catch (NullPointerException e) {
+            logger.warn("Users" + request.getSlackNames() + "isn't found. Received empty list from repository.");
+            throw new UserException("User" + request.getSlackNames() + "found by your request!");
```
Comment (NikolayNN) [inline src/main/java/juja/microservices/users/service/UserService.java]: I think you missed "not" in the exception message

Comment (artem-zhukov) [reply src/main/java/juja/microservices/users/service/UserService.java]: Done

File: src/main/java/juja/microservices/users/service/UserService.java [hunk 2]
```diff
@@ -59,9 +59,15 @@ private UserDTO convertGetAllUsersDto(User user) {
...
-                .map(this::convertGetUuidBySlackDto)
-                .collect(Collectors.toList());
+        List<UserDTO> result;
+        try {
+            result = users.stream()
+                    .map(this::convertGetUuidBySlackDto)
+                    .collect(Collectors.toList());
+        } catch (NullPointerException e) {
+            logger.warn("Users" + request.getSlackNames() + "isn't found. Received empty list from repository.");
+            throw new UserException("User" + request.getSlackNames() + "isn't found by your request!");
```
Comment (NikolayNN) [inline src/main/java/juja/microservices/users/service/UserService.java]: And I think you should use Past Simple Passive, instead Present Simple Passive :)
for example: Users weren't found.

Comment (artem-zhukov) [reply src/main/java/juja/microservices/users/service/UserService.java]: Done :)

File: src/main/java/juja/microservices/users/service/UserService.java [hunk 3]
```diff
@@ -59,9 +59,15 @@ private UserDTO convertGetAllUsersDto(User user) {
...
-        List<UserDTO> result =users.stream()
-                .map(this::convertGetUuidBySlackDto)
-                .collect(Collectors.toList());
+        List<UserDTO> result;
+        try {
+            result = users.stream()
+                    .map(this::convertGetUuidBySlackDto)
+                    .collect(Collectors.toList());
+        } catch (NullPointerException e) {
+            logger.warn("Users" + request.getSlackNames() + "isn't found. Received empty list from repository.");
```
Comment (VadimDyachenko) [inline src/main/java/juja/microservices/users/service/UserService.java]: Некорректное формирование тела логгера. Используй вместо '+ request.getSlackNames() +'  фигурные скобки {}, а после сообщения через запятую указываешь нужные параметры:
logger.warn("Users {} isn't found. Received empty list from repository.", request.getSlackNames());

**Review by VadimDyachenko — COMMENTED** [review]
Оставил комментарий.

File: src/main/java/juja/microservices/users/service/UserService.java (hunk 3 above)
Comment (artem-zhukov) [reply src/main/java/juja/microservices/users/service/UserService.java]: Исправил

File: src/main/java/juja/microservices/users/service/UserService.java [hunk 4]
```diff
@@ -59,9 +59,15 @@ private UserDTO convertGetAllUsersDto(User user) {
...
         logger.debug("Converting received user list to userDTO list");
-        List<UserDTO> result =users.stream()
-                .map(this::convertGetUuidBySlackDto)
-                .collect(Collectors.toList());
+        List<UserDTO> result;
+        try {
+            result = users.stream()
+                    .map(this::convertGetUuidBySlackDto)
+                    .collect(Collectors.toList());
+        } catch (NullPointerException e) {
```
Comment (OlgaKulikova) [inline src/main/java/juja/microservices/users/service/UserService.java:67]: Honestly, I don't like it. Noone catches NullPointerException. If you want to use feature of Java8, look at the Optional class as solution. Or check `if null` with old plain method.

**Review by OlgaKulikova — CHANGES_REQUESTED** [review]

**Review by OlgaKulikova — APPROVED** [review]

**Pull request merged**